shutdown.go

```diff
@@ -49,14 +69,25 @@ type shutdowner struct {
 // In practice this means Shutdowner.Shutdown should not be called from an
 // fx.Invoke, but from a fx.Lifecycle.OnStart hook.
 func (s *shutdowner) Shutdown(opts ...ShutdownOption) error {
-	return s.app.broadcastSignal(_sigTERM)
+	for _, opt := range opts {
+		opt.apply(s)
+	}
+
+	return s.app.broadcastSignal(_sigTERM, s.exitCode)
 }
 
 func (app *App) shutdowner() Shutdowner {
 	return &shutdowner{app: app}
 }
 
-func (app *App) broadcastSignal(signal os.Signal) error {
+func (app *App) broadcastSignal(signal os.Signal, code int) error {
+	return multierr.Combine(
+		app.broadcastDoneSignal(signal),
+		app.broadcastWaitSignal(signal, code),
+	)
```

Comment:
optional/ux improvement:
if both functions return errUnsentSignal,
we should maybe merge those numbers into a single error.
Otherwise we get an error like,

    The following errors occurred:
    - failed to send SIGINT signal to 5 out of 10 channels
    - failed to send SIGINT signal to 7 out of 9 channels

We can keep them separate if we annotate each error with `Wait()` and `Done()`:

    failed to send SIGINT signal to 7 out of 9 Done() channels
    failed to send SIGINT signal to 5 out of 10 Wait() channels